### PR TITLE
[ADP-2367] swap DbPendingTxs layer to new implementation

### PR DESCRIPTION
**Pull request by paolino**
This is the last step of the new submissions store saga.

- swap the dbpendingtx implementation to use the new submissions store
- create migration code
- test migration code 

ADP-2367

### CODE REVIEW

File: lib/wallet/test/unit/Cardano/Wallet/DB/LayerSpec.hs [hunk 1]
```diff
@@ -1357,10 +1376,73 @@ testNewerDatabaseIsNeverModified = withSystemTempFile "db.sql" $ \path _ -> do
...
+          withDBLayer @(SeqState 'Mainnet ShelleyKey) @ShelleyKey
+            nullTracer defaultFieldValues path dummyTimeInterpreter
+                $ \_ -> pure ()
+        testOnCopiedAndMigrated test = fmap snd
+            $ withinCopiedFile dbName $ \path _  -> do
+                performMigrations path
+                test path
+    testOnCopiedAndMigrated testAllPresentTxIdsAreInLedger
+    testOnCopiedAndMigrated allTransactionDataIsRepresentedInTxMeta
+    testOnCopiedAndMigrated testLocalTxSubmissionIsMissing
```
Comment (HeinrichApfelmus) [inline lib/wallet/test/unit/Cardano/Wallet/DB/LayerSpec.hs:1409]: Nice, the function `testOnCopiedAndMigrated` reads very well! 👍

File: lib/wallet/test/unit/Cardano/Wallet/DB/LayerSpec.hs [hunk 2]
```diff
@@ -1357,10 +1376,73 @@ testNewerDatabaseIsNeverModified = withSystemTempFile "db.sql" $ \path _ -> do
...
+            mapM_ (shouldBe' metaIds)
+                [ "tx_in"
+                , "tx_collateral"
+                , "tx_out"
+                , "tx_out_token"
+                , "tx_collateral_out"
+                , "tx_collateral_out_token"
+                , "tx_withdrawal"
+                , "c_b_o_r"
+                ]
```
Comment (HeinrichApfelmus) [inline lib/wallet/test/unit/Cardano/Wallet/DB/LayerSpec.hs:1439]:
```suggestion
                shouldBeSupersetOf mtxs t   = do
                    txs <- runQuery $ selectIds t
                    liftIO $
                        Set.fromList txs `shouldSatisfy`
                            \x -> x `Set.isSubsetOf` (Set.fromList mtxs)
            metaIds  <-  runQuery $ selectIds "tx_meta"
            mapM_ (metaIds `shouldBeSupersetOf`)
                [ "tx_in"
                , "tx_collateral"
                , "tx_out"
                , "tx_out_token"
                , "tx_collateral_out"
                , "tx_collateral_out_token"
                , "tx_withdrawal"
                , "c_b_o_r"
                ]
```

Slightly improved name.

**Review by HeinrichApfelmus — APPROVED** [review]
Excellent, thank you! 😊 Minor comments only, but would be nice to sneak them in.

I'd happy to merge — but one question remains: Does the "LocalTxSubmission pool retries pending transactions" test succeed or fail on this when using the real `DBLayer`? 🤔 I think we may want to have it working — at least in principle (bugs in the tests are ok, bugs in the code are not) — before we merge?

**Comment by paolino** [comment]
Well it's safe to merge IMO. The test is about the re-submitting thread independently on how we store the submissions. In fact it passes the test as it's relying on the specifications for the old encoding. But yeah, let's wait until we sort the test out. We have no rush here. Thanks for reviewing!

**Comment by HeinrichApfelmus** [comment]
After discussion today, it turns out that we need to use `BuiltTx` insted of `SealedTx` in `addTxSubmission_` order to not blow up the test code. We decided to proceed in the following way:

* Keep this PR as it is, no rebase or re-review; but postpone merging it for a little while.
* Make a second PR on top of this one which changes the types as described, and makes the (corrected) unit test pass. As soon as the test passes, we merge this PR, as the second PR witnesses that the test passes.

**Comment by HeinrichApfelmus** [comment]
bors merge

**Comment by paolino** [comment]
bors retry

**Comment by paolino** [comment]
bors retry

**Pull request merged**